includes/embeds/class-amp-core-block-handler.php

```diff
@@ -109,9 +112,105 @@ public function render( $attributes ) {
 			$items_markup
 		);
 
-		if ( ! empty( $attributes['displayAsDropdown'] ) ) {
-			$block_content = sprintf( '<form action="%s" method="get" target="_top" id="%s">%s</form>', esc_url( home_url() ), esc_attr( $form_id ), $block_content );
+		$block_content = sprintf(
+			'<form action="%s" method="get" target="_top" id="%s">%s</form>',
+			esc_url( home_url() ),
+			esc_attr( $form_id ),
+			$block_content
+		);
+
+		return $block_content;
+	}
+
+	/**
+	 * Render archives block when displayAsDropdown.
+	 *
+	 * This replaces disallowed script with the use of on:change for <select>.
+	 *
+	 * @see render_block_core_archives()
+	 *
+	 * @param array $attributes Attributes.
+	 * @return string Rendered.
+	 */
+	public function render_archives_block( $attributes ) {
+		$archives_block = WP_Block_Type_Registry::get_instance()->get_registered( 'core/archives' );
+		if ( empty( $attributes['displayAsDropdown'] ) && isset( $archives_block->original_render_callback ) ) {
+			return call_user_func( $archives_block->original_render_callback, $attributes );
+		}
+
+		$show_post_count = ! empty( $attributes['showPostCounts'] );
+
+		static $block_id = 0;
+		$block_id++;
+
+		$class = 'wp-block-archives';
+
+		if ( isset( $attributes['align'] ) ) {
+			$class .= " align{$attributes['align']}";
+		}
+
+		if ( isset( $attributes['className'] ) ) {
+			$class .= " {$attributes['className']}";
+		}
+
+		$class .= ' wp-block-archives-dropdown';
+
+		$dropdown_id = 'wp-block-categories-dropdown-' . $block_id;
```

Comment:
Is the mixing or `archives` and `categories` correct? (e.g. `wp-block-categories-dropdown-` for tthe Archives widget dropdown)

Reply:
You're right. This is now outdated due to the refactor.